apps/sombrero/reframe_sombrero.py

```diff
@@ -5,38 +5,67 @@
 
 import reframe as rfm
 import reframe.utility.sanity as sn
+from modules.reframe_extras import scaling_config
 
 
 @rfm.simple_test
 class SombreroBenchmark(rfm.RegressionTest):
-    valid_systems = ['*']
-    valid_prog_environs = ['*']
-    build_system = 'Spack'
-    executable = 'sombrero.sh'
-    executable_opts = ['-w', '-s', 'medium']
-    num_tasks = 4
-    time_limit = '10m'
-    variables = {
-        'OMP_NUM_THREADS': '4',
-        'OMP_PLACES': 'cores'
-    }
-    reference = {
-        '*': {
-            'flops': (30, -0.5, 0.5, 'Gflops/seconds'),
+    def __init__(self):
+        self.valid_systems = ['*']
+        self.valid_prog_environs = ['*']
+        self.build_system = 'Spack'
+        self.executable = 'sombrero.sh'
+        self.num_tasks = 2
```

Comment:
`num_tasks` is a bit tricky.  ReFrame uses this parameter to set up the scheduler/launcher with the appropriate number of MPI tasks, so that the job script would have something like
```sh
mpirun -n $num_tasks sombrero.sh ...
```
But the `sombrero.sh` shell driver internally runs the other `sombrero` applications binaries with `mpirun -n ...`, so that there would be two nested `mpirun` calls, which OpenMPI didn't like at all.  That's why in https://github.com/ukri-excalibur/excalibur-tests/commit/a2981df0c6d7325a3ef4138139c8bfd39fd1e9f0 I didn't set `num_tasks` (so it defaults to 1 and the `mpirun` launcher isn't used) and only pass the `-n` argument to `sombrero.sh`

Reply:
In principle `sombrero.sh` is just a convenience wrapper for the actual tests `sombrero1` through `sombrero6`; there's nothing stopping us directly running the individual tests, which would presumably avoid the double `mpirun` call. (The `SUM` calculation would need to be re-done explicitly, but that doesn't seem particularly onerous.)

Reply:
We can make a quick change to the install step in the spack package to make  `sombrero{1..6}` available in the `$PATH`

Reply:
Yes, that'd make it easier to call those commands directly :slightly_smiling_face: